### PR TITLE
Fix to use the validator's pre-image when generating the encryption key.

### LINKED ISSUE #2074: When the encrypted key is provided, it should be changed to use the pre-image of the actual validator.
The encrypted key is used to vote in Votera. 
The pre-image of the validator is not included in the process of creating the encrypted key.
We need to modify to generate an encrypted key using the pre-image associated with that node.

### PR DESCRIPTION
**Pull request by TrustHenry**
The encrypted key is used to vote in Votera.
To prevent this pre-image from being pre-released, add the PreimageRevealPeriod + 1 height.

Related to #2074

### CODE REVIEW

File: source/agora/consensus/EnrollmentManager.d [hunk 1]
```diff
@@ -110,7 +110,7 @@ public class EnrollmentManager
     public EnrollmentPool enroll_pool;
 
     /// Ditto
-    private PreImageCycle cycle;
+    public PreImageCycle cycle;
```
Comment (linked0) [inline source/agora/consensus/EnrollmentManager.d]: The data should be encapsulated. I mean it should be `private`.

File: source/agora/node/admin/AdminInterface.d [hunk 2]
```diff
@@ -183,10 +189,9 @@ public class AdminInterface : NodeControlAPI
     public override string encryptionKeyQR (
         string app, ulong height, out string contentType, out string vary)
     {
-        // TODO: PreImage to be released at that height
-        // Or it can be pre-released, so it can be `height + PreimageRevealPeriod`
-        PreImageInfo pre_image;
-
+        // Gets the pre-image of the requested height.
+        // It can be pre-released, so it add to the `Reveal period + 1`
+        Hash pre_image = this.enroll_man.cycle[Height(height + this.enroll_man.PreimageRevealPeriod + 1)];
```
Comment (linked0) [inline source/agora/node/admin/AdminInterface.d]:
```suggestion
        PreImageInfo preimage;
        this.enroll_man.getNextPreimage(preimage, [Height(height - 1)]);
```
and how about using `preimage.hash` afterward.
If it does not suit for your purpose, I would recommend to add a new function in `EM` for that.

**Review by linked0 — COMMENTED** [review]
Good start.

File: source/agora/node/admin/AdminInterface.d (hunk 2 above)
Comment (TrustHenry) [reply source/agora/node/admin/AdminInterface.d]: The getNextPreimage function can only import pre-images from an enrolled enrollment cycle range.
So I created a function to get a pre-image of the block height.

Comment (TrustHenry) [reply source/agora/node/admin/AdminInterface.d]: @Geod24 ping

File: source/agora/node/admin/AdminInterface.d [hunk 3]
```diff
@@ -319,13 +326,42 @@ public struct EncryptionKey
...
+    auto man = new EnrollmentManager(stateDB, cacheDB, key_pair, new immutable(ConsensusParams)(20));
+    auto adminInterface = new AdminInterface(key_pair, clock, man);
+    string contentType = "image/svg+xml";
+    string vary = "Accept-Encoding";
+    string enc_qr_svg = adminInterface.encryptionKeyQR("Votera", 100, contentType, vary);
+
+    string sample_enc_qr_svg =
+`<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 1.1//EN" "http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd">
+<svg xmlns="http://www.w3.org/2000/svg" version="1.1" viewBox="0 0 500 500">
```
Comment (Geod24) [inline source/agora/node/admin/AdminInterface.d]: This is not really testing anything. I'd rather we remove this and have the test on DevOps side.

Comment (TrustHenry) [reply source/agora/node/admin/AdminInterface.d]: Okay, I removed this test code.

File: source/agora/node/admin/AdminInterface.d [hunk 4]
```diff
@@ -183,9 +189,10 @@ public class AdminInterface : NodeControlAPI
     public override string encryptionKeyQR (
         string app, ulong height, out string contentType, out string vary)
     {
-        // TODO: PreImage to be released at that height
-        // Or it can be pre-released, so it can be `height + PreimageRevealPeriod`
-        PreImageInfo pre_image;
+        // Gets the pre-image of the requested height.
+        // It can be pre-released, so it add to the `Reveal period + 1`
+        Hash pre_image = this.enroll_man.getOurPreimage(
+            Height(height + this.enroll_man.PreimageRevealPeriod + 1));
```
Comment (Geod24) [inline source/agora/node/admin/AdminInterface.d]: No it should just be `height`, the app requesting it should compensate for `PreImageRevealPeriod`

Comment (TrustHenry) [reply source/agora/node/admin/AdminInterface.d]: Okay, I updated.

**Review by Geod24 — APPROVED** [review]

**Pull request merged**